crates/accelerate/src/sparse_observable.rs

```diff
@@ -0,0 +1,1677 @@
+    /// labels for :class:`.Pauli` and :class:`.SparsePauliOp`.
+    ///
+    /// Args:
+    ///     label (str): the dense label.
+    ///
+    /// Examples:
+    ///
+    ///     .. code-block:: python
+    ///         
+    ///         >>> SparseObservable.from_label("IIII+ZI")
+    ///         <SparseObservable with 1 term on 7 qubits: (1+0j)(+_2 Z_1)>
+    ///         >>> label = "IYXZI"
+    ///         >>> pauli = Pauli(label)
+    ///         >>> assert SparseObservable.from_label(label) == SparseObservable.from_pauli(pauli)
+    ///
+    /// See also:
+    ///     :meth:`from_list`
+    ///         A generalization of this method that constructs a sum operator from multiple labels
+    ///         and their corresponding coefficients.
+    #[staticmethod]
+    #[pyo3(name = "from_label", signature = (label, /))]
+    fn py_from_label(label: &str) -> Result<Self, LabelError> {
+        let mut out = Self::zero(label.len() as u32);
+        out.add_dense_label(label, Complex64::new(1.0, 0.0))?;
+        Ok(out)
+    }
+
+    /// Construct an observable from a list of dense labels and coefficients.
+    ///
+    /// This is analagous to :meth:`.SparsePauliOp.from_list`, except it uses
+    /// :ref:`the extended alphabet <sparse-observable-alphabet>` of :class:`.SparseObservable`.  In
+    /// this dense form, you must supply all identities explicitly in each label.
+    ///
+    /// The label must be a sequence of the alphabet ``'IXYZ+-rl01'``.  The label is interpreted
+    /// analagously to a bitstring.  In other words, the right-most letter is associated with qubit
+    /// 0, and so on.  This is the same as the labels for :class:`.Pauli` and
+    /// :class:`.SparsePauliOp`.
+    ///
+    /// Args:
+    ///     iter (list[tuple[str, complex]]): Pairs of labels and their associated coefficients to
+    ///         sum. The labels are interpreted the same way as in :meth:`from_label`.
+    ///     num_qubits (int | None): It is not necessary to specify this if you are sure that
+    ///         ``iter`` is not an empty sequence, since it can be inferred from the label lengths.
+    ///         If ``iter`` may be empty, you must specify this argument to disambiguate how many
+    ///         qubits the observable is for.  If this is given and ``iter`` is not empty, the value
+    ///         must match the label lengths.
+    ///
+    /// Examples:
+    ///
+    ///     Construct an observable from a list of labels of the same length::
+    ///
+    ///         >>> SparseObservable.from_list([
+    ///         ...     ("III++", 1.0),
+    ///         ...     ("II--I", 1.0j),
+    ///         ...     ("I++II", -0.5),
+    ///         ...     ("--III", -0.25j),
+    ///         ... ])
+    ///         <SparseObservable with 4 terms on 5 qubits:
+    ///             (1+0j)(+_1 +_0) + (0+1j)(-_2 -_1) + (-0.5+0j)(+_3 +_2) + (-0-0.25j)(-_4 -_3)>
```

Comment:
I like that the order of operators in the print reflects the qubit indexing 👍🏻  (i.e. that it is `+_1 +_0` instead of `+_0 +_1` -- ok the difference would be more visible if the bit terms were different haha)

Reply:
Ha, thanks - that was completely deliberate.  I thought it's hard enough to understand what's going on with the sparsity as things are, so I wanted to be really crystal clear.

I documented somewhere that `from_list` is equivalent to `from_sparse_list` if all the coeffs are `reversed(range(num_qubits))` as well, so these two are the same:
```python
SparseObservable.from_list([("IXIZ", 1j), ("+-rl", 0.5)])
SparseObservable.from_sparse_list([("IXIZ", (3, 2, 1, 0), 1j), ("+-rl", (3, 2, 1, 0), 0.5)])
```
to try and drive the point home.

Reply:
Yeah that is actually a nice way of writing the sparse labels, maybe we should update the `SparsePauliOp` docstring to add this example 🤔

Reply:
tbh, the `SparsePauliOp` documentation needs a whole overhaul - it's really unclear for how core the class is to the V1 and V2 primitives.  I was trying to get ahead of the game with this operator.  We also should overhaul the `quantum_info` module page in a similar vein, and much of the classes.  Abby and the docs team have got a bunch of open issues about that in #13236.